File: src/main/java/com/sudosoo/takeiteasy/service/PostService.java
package com.sudosoo.takeiteasy.service;

import com.sudosoo.takeiteasy.dto.post.*;
import com.sudosoo.takeiteasy.entity.Post;
import org.springframework.data.domain.Pageable;

import java.io.IOException;
import java.util.List;
import java.util.concurrent.ExecutionException;
import java.util.concurrent.TimeoutException;

public interface PostService {
    PostResponseDto create(CreatePostRequestDto createPostRequestDto) throws ExecutionException, InterruptedException, IOException, TimeoutException;
    Post getByPostId(Long postId);
    PostDetailResponseDto getPostDetailByPostId(Long postId, Pageable pageRequest);
    List<PostTitleOnlyResponseDto> getPaginationPost(Pageable pageable);
    Post createBatchPosts(int count);
    PostResponseDto redisTest(PostRequestDto requestDto);
}

File: takeItEasy-application/src/main/kotlin/com/sudosoo/takeItEasy/service/PostService.kt
package com.sudosoo.takeItEasy.service

import com.sudosoo.takeItEasy.dto.post.*
import com.sudosoo.takeiteasy.entity.Post
import org.springframework.data.domain.Pageable
import java.io.IOException
import java.util.concurrent.ExecutionException
import java.util.concurrent.TimeoutException

interface PostService {
    @Throws(ExecutionException::class, InterruptedException::class, IOException::class, TimeoutException::class)
    fun create(createPostRequestDto: CreatePostRequestDto): TestPostResponseDto
    fun getByPostId(postId: Long): Post
    fun getPostDetailByPostId(postId: Long, pageRequest: Pageable): PostDetailResponseDto
    fun getPaginationPost(pageable: Pageable): List<PostTitleOnlyResponseDto>
    fun createBatchPosts(count: Int): Post
    fun redisTest(requestDto: PostRequestDto): TestPostResponseDto
    fun allPost() : List<TestPostResponseDto>
}
